users/tests.py

```diff
@@ -60,6 +61,51 @@ def test_valid_otp_should_pass_new_user(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
 
+class ThirdPartyAuthTestCase(BaseTestCase):
+    """Tests the third party auth functionality"""
+
+    def setUp(self):
+        super().setUp()
+        # unset client credentials token so that the subsequent API calls goes as guest
+        self.client.credentials()
+
+    def test_cannot_authenticate_without_all_required_keys(self):
+        # keeping only 2 out of the 3 required auth keys
+        payload = {"auth_type": "abcd", "unique_id": "1234"}
+        response = self.client.post(
+            reverse("convert_third_party_access_token"), payload
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.data["detail"], "access_token not provided")
+
+    def test_guest_can_authenticate_using_third_party(self):
```

Comment:
one more test for user authenticating through 3rd party when that user already exists.